# CMU Movies Summary Corpus

- Team: ArabsDataScientists2024
- Authors: Zaynab, Lylia, Ali, Christian, Yassin

---

## Tasks

1. **Select Project & Initial Analyses**:
   1. Agree on a project proposal with team members.
   2. Perform initial analyses to verify feasibility of the proposed project, including any additional data.
   3. Acquaint yourself with the provided data, preprocess it, and perform descriptive statistics.

2. **Pipeline & Data Description**:
   1. Create a pipeline for data handling and preprocessing, documented in the notebook.
   2. Describe the relevant aspects of the data, including:
      1. Handling the size of the data.
      2. Understanding the data (formats, distributions, missing values, correlations, etc.).
      3. Considering data enrichment, filtering, and transformation according to project needs.
   3. Develop a plan for methods to be used, with essential mathematical details.
   4. Outline a plan for analysis and communication, discussing alternative approaches considered.

3. **GitHub Repository & Deliverables**:
   1. Create a public GitHub repository named `ada-2023-project-<team>` under the `epfl-ada` GitHub organization. ✅
   2. Ensure the repository contains:
      1. **README.md** file with:
         1. **Title**: Project title.
         2. **Abstract**: 150-word description of the project idea, goals, and motivation.
         3. **Research Questions**: List of research questions to address.
         4. **Proposed Additional Datasets**: Description of additional datasets, expected management, and feasibility analysis.
         5. **Methods**: Methods to be used in the project.
         6. **Proposed Timeline**: Timeline for the project.
         7. **Organization within the Team**: Internal milestones leading to Milestone P3.
         8. **Questions for TAs (optional)**: Any questions for the teaching assistants.
      2. **Code for Initial Analyses**: Structured code for initial analyses and data handling pipelines.
      3. **Notebook** presenting initial results, including:
         1. Main results and descriptive analysis.
         2. External scripts/modules for implementing core logic, to be called from the notebook.

---


## Table of Contents
- [1. Zaynab's part](##Zaynab's-part)
- [2. Lylia's part](##Lylia's-part)
- [3. Ali's part](##Ali's-part)
- [4. Cristians's part](##Christian's-part)
- [5. Yassin's part](##Yassin's-part)

---

### Library importation

In [2]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
import kagglehub

### data importation

In [3]:
DATA_PATH='./data/MovieSummaries/'

### movie metadata

In [4]:
movie_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'MovieName', 'ReleaseDate', 
    'BoxOfficeRevenue', 'Runtime', 'Languages', 'Countries', 'Genres'
]


movie_metadata = pd.read_csv(DATA_PATH+'movie.metadata.tsv', sep='\t', names=movie_columns)

movie_metadata

WikipediaMovieID FreebaseMovieID  \
0                975900       /m/03vyhn   
1               3196793       /m/08yl5d   
2              28463795      /m/0crgdbh   
3               9363483      /m/0285_cd   
4                261236       /m/01mrr1   
...                 ...             ...   
81736          35228177      /m/0j7hxnt   
81737          34980460      /m/0g4pl34   
81738           9971909      /m/02pygw1   
81739            913762       /m/03pcrp   
81740          12476867      /m/02w7zz8   

                                               MovieName ReleaseDate  \
0                                         Ghosts of Mars  2001-08-24   
1      Getting Away with Murder: The JonBenét Ramsey ...  2000-02-16   
2                                            Brun bitter        1988   
3                                       White Of The Eye        1987   
4                                      A Woman in Flames        1983   
...                                                  ...         ...   
81736                           Mermaids: The Body Found  2011-03-19   
81737                                            Knuckle  2011-01-21   
81738                                  Another Nice Mess  1972-09-22   
81739  The Super Dimension Fortress Macross II: Lover...  1992-05-21   
81740                                            Spliced        2002   

       BoxOfficeRevenue  Runtime                           Languages  \
0            14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1                   NaN     95.0  {"/m/02h40lc": "English Language"}   
2                   NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3                   NaN    110.0  {"/m/02h40lc": "English Language"}   
4                   NaN    106.0   {"/m/04306rv": "German Language"}   
...                 ...      ...                                 ...   
81736               NaN    120.0  {"/m/02h40lc": "English Language"}   
81737               NaN     96.0  {"/m/02h40lc": "English Language"}   
81738               NaN     66.0  {"/m/02h40lc": "English Language"}   
81739               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               Countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  Genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

### character metadata

In [5]:
character_columns = [
    'WikipediaMovieID', 'FreebaseMovieID', 'ReleaseDate', 'CharacterName',
    'ActorDOB', 'ActorGender', 'ActorHeight', 'ActorEthnicity', 
    'ActorName', 'ActorAgeAtRelease', 'FreebaseCharacterActorMapID',
    'FreebaseCharacterID', 'FreebaseActorID'
]

character_metadata = pd.read_csv(DATA_PATH+'character.metadata.tsv', sep='\t', names=character_columns)


character_metadata

WikipediaMovieID FreebaseMovieID ReleaseDate  \
0                 975900       /m/03vyhn  2001-08-24   
1                 975900       /m/03vyhn  2001-08-24   
2                 975900       /m/03vyhn  2001-08-24   
3                 975900       /m/03vyhn  2001-08-24   
4                 975900       /m/03vyhn  2001-08-24   
...                  ...             ...         ...   
450664            913762       /m/03pcrp  1992-05-21   
450665            913762       /m/03pcrp  1992-05-21   
450666          28308153      /m/0cp05t9        1957   
450667          28308153      /m/0cp05t9        1957   
450668          28308153      /m/0cp05t9        1957   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                      Elensh     1970-05           F          NaN   
450665                      Hibiki  1965-04-12           M          NaN   
450666                         NaN  1941-11-18           M        1.730   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN   

       ActorEthnicity           ActorName  ActorAgeAtRelease  \
0                 NaN      Wanda De Jesus               42.0   
1          /m/044038p  Natasha Henstridge               27.0   
2             /m/0x67            Ice Cube               32.0   
3                 NaN       Jason Statham               33.0   
4                 NaN         Clea DuVall               23.0   
...               ...                 ...                ...   
450664            NaN  Dorothy Elias-Fahn                NaN   
450665            NaN       Jonathan Fahn               27.0   
450666      /m/02w7gg      David Hemmings               15.0   
450667            NaN    Roberta Paterson                NaN   
450668            NaN         John Rogers                NaN   

       FreebaseCharacterActorMapID FreebaseCharacterID FreebaseActorID  
0                       /m/0bgchxw          /m/0bgcj3x      /m/03wcfv7  
1                        /m/0jys3m          /m/0bgchn4       /m/0346l4  
2                        /m/0jys3g          /m/0bgchn_      /m/01vw26l  
3                       /m/02vchl6          /m/0bgchnq       /m/034hyc  
4                       /m/02vbb3r          /m/0bgchp9       /m/01y9xg  
...                            ...                 ...             ...  
450664                  /m/0kr406c          /m/0kr406h       /m/0b_vcv  
450665                  /m/0kr405_          /m/0kr4090       /m/0bx7_j  
450666                  /m/0g8ngmc                 NaN       /m/022g44  
450667                  /m/0g8ngmj                 NaN      /m/0g8ngmm  
450668                  /m/0g8ngmw                 NaN      /m/0btz19d  

[450669 rows x 13 columns]

### plot summaries

In [6]:
plot_columns = ['WikipediaMovieID', 'PlotSummary']

plot_summaries = pd.read_csv(DATA_PATH+'plot_summaries.txt', sep='\t', names=plot_columns)

plot_summaries


WikipediaMovieID                                        PlotSummary
0              23890098  Shlykov, a hard-working taxi driver and Lyosha...
1              31186339  The nation of Panem consists of a wealthy Capi...
2              20663735  Poovalli Induchoodan  is sentenced for six yea...
3               2231378  The Lemon Drop Kid , a New York City swindler,...
4                595909  Seventh-day Adventist Church pastor Michael Ch...
...                 ...                                                ...
42298          34808485  The story is about Reema , a young Muslim scho...
42299           1096473  In 1928 Hollywood, director Leo Andreyev  look...
42300          35102018  American Luthier focuses on Randy Parsons’ tra...
42301           8628195  Abdur Rehman Khan , a middle-aged dry fruit se...
42302           6040782  1940 - Operation Dynamo has just taken place. ...

[42303 rows x 2 columns]

### name clusters

In [7]:
name_clusters_columns = ['FreebaseCharacterActorMapID', 'CharacterName']

name_clusters = pd.read_csv(DATA_PATH+'name.clusters.txt', sep='\t', names=name_clusters_columns)

name_clusters


FreebaseCharacterActorMapID CharacterName
0                  Stuart Little     /m/0k3w9c
1                  Stuart Little     /m/0k3wcx
2                  Stuart Little     /m/0k3wbn
3                       John Doe     /m/0jyg35
4                       John Doe     /m/0k2_zn
...                          ...           ...
2661                  John Rolfe     /m/0k5_ql
2662                  John Rolfe    /m/02vd6vs
2663             Elizabeth Swann     /m/0k1xvz
2664             Elizabeth Swann     /m/0k1x_d
2665             Elizabeth Swann     /m/0k1xxy

[2666 rows x 2 columns]

### TV tropes clusters

In [8]:
tvtropes_columns = ['FreebaseCharacterActorMapID', 'CharacterType']

tvtropes_clusters = pd.read_csv(DATA_PATH+'tvtropes.clusters.txt', sep='\t', names=tvtropes_columns)

tvtropes_clusters


FreebaseCharacterActorMapID  \
0       absent_minded_professor   
1       absent_minded_professor   
2       absent_minded_professor   
3       absent_minded_professor   
4       absent_minded_professor   
..                          ...   
496                   young_gun   
497                   young_gun   
498                   young_gun   
499                   young_gun   
500                   young_gun   

                                         CharacterType  
0    {"char": "Professor Philip Brainard", "movie":...  
1    {"char": "Professor Keenbean", "movie": "Richi...  
2    {"char": "Dr. Reinhardt Lane", "movie": "The S...  
3    {"char": "Dr. Harold Medford", "movie": "Them!...  
4    {"char": "Daniel Jackson", "movie": "Stargate"...  
..                                                 ...  
496  {"char": "Morgan Earp", "movie": "Tombstone", ...  
497  {"char": "Colorado Ryan", "movie": "Rio Bravo"...  
498  {"char": "Tom Sawyer", "movie": "The League of...  
499  {"char": "William H. 'Billy the Kid' Bonney", ...  
500  {"char": "Jake", "movie": "Silverado", "id": "...  

[501 rows x 2 columns]

---

## Zaynab's part

---
## Lylia's part

---
## Ali's part

---
## Christian's part

---
## Yassin's part

### Sites with additional data:

dataset with physique : https://arxiv.org/html/2407.03486v1

celeb A: https://mmlab.ie.cuhk.edu.hk/projects/CelebA.html

https://www.kaggle.com/datasets/vishesh1412/celebrity-face-image-dataset

http://www.stat.ucla.edu/~vlew/stat10/archival/SP01/handouts/celeb.html

IMDB : https://datasets.imdbws.com/

boxofficeojo: https://www.kaggle.com/datasets/eliasdabbas/boxofficemojo-alltime-domestic-data


In [49]:
movie_metadata['year']=movie_metadata['ReleaseDate'].str.extract(r'(\d{4})')
movie_metadata['year'] = movie_metadata['year'].astype(float)
print(len(movie_metadata))
print(movie_metadata['year'].isna().sum())

81741
6902


In [50]:
movie_metadata['year'].describe()

count    74839.000000
mean      1977.476530
std         29.101536
min       1010.000000
25%       1956.000000
50%       1985.000000
75%       2004.000000
max       2016.000000
Name: year, dtype: float64

In [ ]:
rows_with_1010 = movie_metadata[movie_metadata['year'] == 1010]

print(rows_with_1010)
print(rows_with_1010.index)

       WikipediaMovieID FreebaseMovieID       MovieName ReleaseDate  \
62836          29666067      /m/0fphzrf  Hunting Season  1010-12-02   

       BoxOfficeRevenue  Runtime  \
62836        12160978.0    140.0   

                                               Languages  \
62836  {"/m/02hwyss": "Turkish Language", "/m/02h40lc...   

                     Countries  \
62836  {"/m/01znc_": "Turkey"}   

                                                  Genres    year  
62836  {"/m/0lsxr": "Crime Fiction", "/m/02n4kr": "My...  1010.0  
Index([62836], dtype='int64')


In [53]:
movie_metadata.at[62836, 'year'] = 2010 

correcting this error, movie date was 1010 when is supposed to be 2010

In [54]:
print(movie_metadata['year'].describe())

count    74839.000000
mean      1977.489892
std         28.886090
min       1888.000000
25%       1956.000000
50%       1985.000000
75%       2004.000000
max       2016.000000
Name: year, dtype: float64


The first movie of our datset was released in 1888 and the latest in 2016

In [55]:
movie_metadata['BoxOfficeRevenue'].isna().sum()/len(movie_metadata['BoxOfficeRevenue'])

0.8972241592346558

In [56]:
character_metadata['ActorEthnicity'].isna().sum()/len(character_metadata['ActorEthnicity']) 

0.7646654196317031

In [57]:
character_metadata['ActorHeight'].isna().sum()/len(character_metadata['ActorHeight'])

0.6564573999986687

In [58]:
movie_metadata['Countries'].unique()

array(['{"/m/09c7w0": "United States of America"}',
       '{"/m/05b4w": "Norway"}', '{"/m/07ssc": "United Kingdom"}', ...,
       '{"/m/0f8l9c": "France", "/m/06mzp": "Switzerland", "/m/0h3y": "Algeria", "/m/0345h": "Germany"}',
       '{"/m/014tss": "Kingdom of Great Britain", "/m/03_3d": "Japan", "/m/02jx1": "England", "/m/07ssc": "United Kingdom", "/m/0345h": "Germany"}',
       '{"/m/06mzp": "Switzerland", "/m/03rjj": "Italy", "/m/082fr": "West Germany", "/m/03f2w": "German Democratic Republic"}'],
      dtype=object)

In [59]:
movie_metadata['Languages'].unique()

array(['{"/m/02h40lc": "English Language"}',
       '{"/m/05f_3": "Norwegian Language"}',
       '{"/m/04306rv": "German Language"}', ...,
       '{"/m/03k50": "Hindi Language", "/m/064r7fk": "Standard Tibetan", "/m/02h40lc": "English Language"}',
       '{"/m/06nm1": "Spanish Language", "/m/04306rv": "German Language", "/m/02h40lc": "English Language", "/m/02ztjwg": "Hungarian language"}',
       '{"/m/02bjrlw": "Italian Language", "/m/02h40lc": "English Language", "/m/05f_3": "Norwegian Language"}'],
      dtype=object)

In [ ]:
# # Download latest version
# path = kagglehub.dataset_download("stefanoleone992/rotten-tomatoes-movies-and-critic-reviews-dataset")

# print("Path to dataset files:", path)
# path = kagglehub.dataset_download("patkle/metacritic-scores-for-games-movies-tv-and-music")

# print("Path to dataset files:", path)
# path = kagglehub.dataset_download("eliasdabbas/boxofficemojo-alltime-domestic-data")

# print("Path to dataset files:", path)

# path = kagglehub.dataset_download("martinmraz07/oscar-movies")

# print("Path to dataset files:", path)

# path = kagglehub.dataset_download("unanimad/the-oscar-award")

# print("Path to dataset files:", path)

In [20]:
ratings_IMDB=pd.read_csv('data/IMDB/title.ratings.tsv.gz', sep='\t')
ratings_IMDB

tconst  averageRating  numVotes
0        tt0000001            5.7      2096
1        tt0000002            5.6       282
2        tt0000003            6.5      2115
3        tt0000004            5.4       182
4        tt0000005            6.2      2845
...            ...            ...       ...
1495756  tt9916730            7.0        12
1495757  tt9916766            7.1        24
1495758  tt9916778            7.2        37
1495759  tt9916840            6.9        11
1495760  tt9916880            7.9         9

[1495761 rows x 3 columns]

In [27]:
akas_IMDB=pd.read_csv('data/IMDB/title.akas.tsv.gz', sep='\t')

akas_IMDB

titleId  ordering                      title region language  \
0         tt0000001         1                 Carmencita     \N       \N   
1         tt0000001         2                 Carmencita     DE       \N   
2         tt0000001         3                 Carmencita     US       \N   
3         tt0000001         4  Carmencita - spanyol tánc     HU       \N   
4         tt0000001         5                 Καρμενσίτα     GR       \N   
...             ...       ...                        ...    ...      ...   
50301474  tt9916852         7                エピソード #3.20     JP       ja   
50301475  tt9916852         8             Episodio #3.20     ES       es   
50301476  tt9916856         1                   The Wind     \N       \N   
50301477  tt9916856         2                   The Wind     DE       \N   
50301478  tt9916880         1  Horrid Henry Knows It All     \N       \N   

                types     attributes  isOriginalTitle  
0            original             \N                1  
1                  \N  literal title                0  
2         imdbDisplay             \N                0  
3         imdbDisplay             \N                0  
4         imdbDisplay             \N                0  
...               ...            ...              ...  
50301474           \N             \N                0  
50301475           \N             \N                0  
50301476     original             \N                1  
50301477  imdbDisplay             \N                0  
50301478     original             \N                1  

[50301479 rows x 8 columns]

In [28]:
characters_IMDB=pd.read_csv('data/IMDB/name.basics.tsv.gz', sep='\t')

characters_IMDB

nconst         primaryName birthYear deathYear  \
0         nm0000001        Fred Astaire      1899      1987   
1         nm0000002       Lauren Bacall      1924      2014   
2         nm0000003     Brigitte Bardot      1934        \N   
3         nm0000004        John Belushi      1949      1982   
4         nm0000005      Ingmar Bergman      1918      2007   
...             ...                 ...       ...       ...   
13926614  nm9993714   Romeo del Rosario        \N        \N   
13926615  nm9993716       Essias Loberg        \N        \N   
13926616  nm9993717  Harikrishnan Rajan        \N        \N   
13926617  nm9993718         Aayush Nair        \N        \N   
13926618  nm9993719          Andre Hill        \N        \N   

                            primaryProfession  \
0                actor,miscellaneous,producer   
1          actress,soundtrack,archive_footage   
2           actress,music_department,producer   
3               actor,writer,music_department   
4                       writer,director,actor   
...                                       ...   
13926614  animation_department,art_department   
13926615                                   \N   
13926616                      cinematographer   
13926617                      cinematographer   
13926618                                   \N   

                                   knownForTitles  
0         tt0050419,tt0072308,tt0053137,tt0027125  
1         tt0037382,tt0075213,tt0117057,tt0038355  
2         tt0057345,tt0049189,tt0056404,tt0054452  
3         tt0072562,tt0077975,tt0080455,tt0078723  
4         tt0050986,tt0069467,tt0083922,tt0050976  
...                                           ...  
13926614          tt11657662,tt14069590,tt2455546  
13926615                                       \N  
13926616                                tt8736744  
13926617                                tt8736744  
13926618                                       \N  

[13926619 rows x 6 columns]

<!-- To use Tweepy for downloading Twitter follower counts for a list of celebrities, you’ll need to set up a Twitter Developer account, create an application to get access credentials, and then use Tweepy to interact with the Twitter API. Here’s a step-by-step guide:

1. Set Up Twitter Developer Account and Application
Go to the Twitter Developer Portal, create an account, and apply for API access.
Create a new app within your developer account. Twitter will provide you with an API key, API secret key, Access token, and Access token secret.
2. Install Tweepy
Install Tweepy using pip:

bash
Copy code
pip install tweepy
3. Authenticate with the Twitter API
In your Python script, you can authenticate with Twitter using the credentials you got from the developer portal.

python
Copy code
import tweepy

# Replace these with your own credentials
API_KEY = 'your_api_key'
API_SECRET_KEY = 'your_api_secret_key'
ACCESS_TOKEN = 'your_access_token'
ACCESS_TOKEN_SECRET = 'your_access_token_secret'

# Authenticate with Twitter
auth = tweepy.OAuthHandler(API_KEY, API_SECRET_KEY)
auth.set_access_token(ACCESS_TOKEN, ACCESS_TOKEN_SECRET)
api = tweepy.API(auth)
4. Get Follower Counts for Celebrities
Create a list of Twitter handles for the celebrities you want to track and fetch their follower counts using Tweepy.

python
Copy code
# List of Twitter usernames for celebrities
celebrity_usernames = ['elonmusk', 'rihanna', 'katyperry', 'TheRock']  # Replace with actual usernames

# Loop through each username and get follower count
follower_data = {}
for username in celebrity_usernames:
    try:
        user = api.get_user(screen_name=username)
        follower_data[username] = user.followers_count
        print(f"{username}: {user.followers_count} followers")
    except tweepy.TweepError as e:
        print(f"Error fetching data for {username}: {e}")

# Optional: Save to a file
import csv

with open('celebrity_followers.csv', 'w', newline='') as file:
    writer = csv.writer(file)
    writer.writerow(["Username", "Follower Count"])
    for username, followers in follower_data.items():
        writer.writerow([username, followers])
5. Run the Script Regularly (Optional)
If you’d like to track follower count changes over time, you can run this script daily, weekly, etc., and save the data to a CSV file or database.

Notes and Tips
Rate Limits: Twitter has rate limits, so if you have a large list of celebrities, you may hit the limit. Tweepy will handle some rate limiting for you, but you may need to wait a few minutes if you have a long list.
Error Handling: Some accounts may be private, suspended, or may change usernames. Use error handling (as shown above) to skip over these cases.
This approach will give you up-to-date follower counts for each celebrity and can easily be expanded or automated as needed. Let me know if you need help with any specific part! -->

In [30]:
title_basics_IMDB=pd.read_csv('data/IMDB/title.basics.tsv.gz', sep='\t')

title_basics_IMDB

C:\Users\Yass La Menace\AppData\Local\Temp\ipykernel_22060\4251460313.py:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  title_basics_IMDB=pd.read_csv('data/IMDB/title.basics.tsv.gz', sep='\t')


tconst  titleType               primaryTitle  \
0         tt0000001      short                 Carmencita   
1         tt0000002      short     Le clown et ses chiens   
2         tt0000003      short               Poor Pierrot   
3         tt0000004      short                Un bon bock   
4         tt0000005      short           Blacksmith Scene   
...             ...        ...                        ...   
11217428  tt9916848  tvEpisode              Episode #3.17   
11217429  tt9916850  tvEpisode              Episode #3.19   
11217430  tt9916852  tvEpisode              Episode #3.20   
11217431  tt9916856      short                   The Wind   
11217432  tt9916880  tvEpisode  Horrid Henry Knows It All   

                      originalTitle isAdult startYear endYear runtimeMinutes  \
0                        Carmencita       0      1894      \N              1   
1            Le clown et ses chiens       0      1892      \N              5   
2                    Pauvre Pierrot       0      1892      \N              5   
3                       Un bon bock       0      1892      \N             12   
4                  Blacksmith Scene       0      1893      \N              1   
...                             ...     ...       ...     ...            ...   
11217428              Episode #3.17       0      2009      \N             \N   
11217429              Episode #3.19       0      2010      \N             \N   
11217430              Episode #3.20       0      2010      \N             \N   
11217431                   The Wind       0      2015      \N             27   
11217432  Horrid Henry Knows It All       0      2014      \N             10   

                              genres  
0                  Documentary,Short  
1                    Animation,Short  
2           Animation,Comedy,Romance  
3                    Animation,Short  
4                       Comedy,Short  
...                              ...  
11217428         Action,Drama,Family  
11217429         Action,Drama,Family  
11217430         Action,Drama,Family  
11217431                       Short  
11217432  Adventure,Animation,Comedy  

[11217433 rows x 9 columns]

In [31]:
crew_IMDB=pd.read_csv('data/IMDB/title.crew.tsv.gz', sep='\t')

In [37]:
crew_IMDB

tconst   directors  \
0         tt0000001   nm0005690   
1         tt0000002   nm0721526   
2         tt0000003   nm0721526   
3         tt0000004   nm0721526   
4         tt0000005   nm0005690   
...             ...         ...   
10553131  tt9916848   nm1485677   
10553132  tt9916850   nm1485677   
10553133  tt9916852   nm1485677   
10553134  tt9916856  nm10538645   
10553135  tt9916880   nm0996406   

                                                    writers  
0                                                        \N  
1                                                        \N  
2                                                        \N  
3                                                        \N  
4                                                        \N  
...                                                     ...  
10553131  nm9187127,nm1485677,nm9826385,nm9299459,nm1628284  
10553132            nm9187127,nm1485677,nm9826385,nm1628284  
10553133  nm9187127,nm1485677,nm9826385,nm9299459,nm1628284  
10553134                                          nm6951431  
10553135                                nm1482639,nm2586970  

[10553136 rows x 3 columns]

In [33]:
episode_IMDB=pd.read_csv('data/IMDB/title.episode.tsv.gz', sep='\t')
principals_IMDB=pd.read_csv('data/IMDB/title.principals.tsv.gz', sep='\t')

In [38]:
episode_IMDB

tconst parentTconst seasonNumber episodeNumber
0        tt0031458   tt32857063           \N            \N
1        tt0041951    tt0041038            1             9
2        tt0042816    tt0989125            1            17
3        tt0042889    tt0989125           \N            \N
4        tt0043426    tt0040051            3            42
...            ...          ...          ...           ...
8614779  tt9916846    tt1289683            3            18
8614780  tt9916848    tt1289683            3            17
8614781  tt9916850    tt1289683            3            19
8614782  tt9916852    tt1289683            3            20
8614783  tt9916880    tt0985991            4             2

[8614784 rows x 4 columns]

In [39]:
principals_IMDB

tconst  ordering     nconst         category  \
0         tt0000001         1  nm1588970             self   
1         tt0000001         2  nm0005690         director   
2         tt0000001         3  nm0005690         producer   
3         tt0000001         4  nm0374658  cinematographer   
4         tt0000002         1  nm0721526         director   
...             ...       ...        ...              ...   
88937148  tt9916880        17  nm0996406         director   
88937149  tt9916880        18  nm1482639           writer   
88937150  tt9916880        19  nm2586970           writer   
88937151  tt9916880        20  nm1594058         producer   
88937152  tt9916880        21  nm1482639         producer   

                              job characters  
0                              \N   ["Self"]  
1                              \N         \N  
2                        producer         \N  
3         director of photography         \N  
4                              \N         \N  
...                           ...        ...  
88937148       principal director         \N  
88937149                       \N         \N  
88937150                    books         \N  
88937151                 producer         \N  
88937152                 producer         \N  

[88937153 rows x 6 columns]

In [44]:
character_metadata['ActorName'].value_counts()

ActorName
Mel Blanc             791
Mithun Chakraborty    328
Oliver Hardy          299
Mohanlal              234
Moe Howard            225
                     ... 
Jan van der Horst       1
Irma Lozin              1
Johan Gildemeijer       1
Jan Feith               1
Roberta Paterson        1
Name: count, Length: 134078, dtype: int64

In [63]:
character_metadata['year']=character_metadata['ReleaseDate'].str.extract(r'(\d{4})')
character_metadata['year'] = character_metadata['year'].astype(float)
print(len(character_metadata))
print(character_metadata['year'].isna().sum())

450669
9995


In [65]:
rows_character_1010 =character_metadata[character_metadata['year']==1010]
print(rows_character_1010)

       WikipediaMovieID FreebaseMovieID ReleaseDate   CharacterName  \
67624          29666067      /m/0fphzrf  1010-12-02         Kamuran   
67625          29666067      /m/0fphzrf  1010-12-02          Ferman   
67626          29666067      /m/0fphzrf  1010-12-02           Idris   
67627          29666067      /m/0fphzrf  1010-12-02           Hasan   
67628          29666067      /m/0fphzrf  1010-12-02          Battal   
67629          29666067      /m/0fphzrf  1010-12-02           Asiye   
67630          29666067      /m/0fphzrf  1010-12-02       Asit Omer   
67631          29666067      /m/0fphzrf  1010-12-02           Hatun   
67632          29666067      /m/0fphzrf  1010-12-02          Müslüm   
67633          29666067      /m/0fphzrf  1010-12-02      Murat Önes   
67634          29666067      /m/0fphzrf  1010-12-02           Hilal   
67635          29666067      /m/0fphzrf  1010-12-02         Cevriye   
67636          29666067      /m/0fphzrf  1010-12-02        Müzeyyen   
67637 

In [64]:
character_metadata['year'].describe()

count    440674.000000
mean       1984.489929
std          25.889522
min        1010.000000
25%        1969.000000
50%        1994.000000
75%        2005.000000
max        2016.000000
Name: year, dtype: float64

In [66]:
character_metadata.loc[character_metadata['year'] == 1010, 'year'] = 2010


In [67]:
character_metadata['year'].describe()

count    440674.000000
mean       1984.523968
std          25.257949
min        1888.000000
25%        1969.000000
50%        1994.000000
75%        2005.000000
max        2016.000000
Name: year, dtype: float64

In [68]:
rows_character_1010 =character_metadata[character_metadata['year']==1010]
print(rows_character_1010)

Empty DataFrame
Columns: [WikipediaMovieID, FreebaseMovieID, ReleaseDate, CharacterName, ActorDOB, ActorGender, ActorHeight, ActorEthnicity, ActorName, ActorAgeAtRelease, FreebaseCharacterActorMapID, FreebaseCharacterID, FreebaseActorID, year]
Index: []


In [69]:
print(movie_metadata['WikipediaMovieID'].isna().sum()/(len(movie_metadata['WikipediaMovieID'])))
print(movie_metadata['FreebaseMovieID'].isna().sum()/(len(movie_metadata['FreebaseMovieID'])))

0.0
0.0


### merging understanding

In [22]:
import pandas as pd

data1 = {
  "name": ["Sally", "Mary", "John"],
  "age": [50, 40, 30]
}

data2 = {
  "name": ["Sally", "Peter", "Micky"],
  "age": [77, 44, 22]
}

df1 = pd.DataFrame(data1)
df2 = pd.DataFrame(data2)

newdf = df1.merge(df2, how='right', on='name')
newdf

name  age_x  age_y
0  Sally   50.0     77
1  Peter    NaN     44
2  Micky    NaN     22

In [23]:
merged_data=pd.merge(movie_metadata,character_metadata,how='inner',on='WikipediaMovieID')
merged_data

WikipediaMovieID FreebaseMovieID_x       MovieName ReleaseDate_x  \
0                 975900         /m/03vyhn  Ghosts of Mars    2001-08-24   
1                 975900         /m/03vyhn  Ghosts of Mars    2001-08-24   
2                 975900         /m/03vyhn  Ghosts of Mars    2001-08-24   
3                 975900         /m/03vyhn  Ghosts of Mars    2001-08-24   
4                 975900         /m/03vyhn  Ghosts of Mars    2001-08-24   
...                  ...               ...             ...           ...   
450664          12476867        /m/02w7zz8         Spliced          2002   
450665          12476867        /m/02w7zz8         Spliced          2002   
450666          12476867        /m/02w7zz8         Spliced          2002   
450667          12476867        /m/02w7zz8         Spliced          2002   
450668          12476867        /m/02w7zz8         Spliced          2002   

        BoxOfficeRevenue  Runtime                           Languages  \
0             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
2             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
3             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
4             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
...                  ...      ...                                 ...   
450664               NaN     86.0  {"/m/02h40lc": "English Language"}   
450665               NaN     86.0  {"/m/02h40lc": "English Language"}   
450666               NaN     86.0  {"/m/02h40lc": "English Language"}   
450667               NaN     86.0  {"/m/02h40lc": "English Language"}   
450668               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                        Countries  \
0       {"/m/09c7w0": "United States of America"}   
1       {"/m/09c7w0": "United States of America"}   
2       {"/m/09c7w0": "United States of America"}   
3       {"/m/09c7w0": "United States of America"}   
4       {"/m/09c7w0": "United States of America"}   
...                                           ...   
450664                    {"/m/0d060g": "Canada"}   
450665                    {"/m/0d060g": "Canada"}   
450666                    {"/m/0d060g": "Canada"}   
450667                    {"/m/0d060g": "Canada"}   
450668                    {"/m/0d060g": "Canada"}   

                                                   Genres FreebaseMovieID_y  \
0       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...         /m/03vyhn   
1       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...         /m/03vyhn   
2       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...         /m/03vyhn   
3       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...         /m/03vyhn   
4       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...         /m/03vyhn   
...                                                   ...               ...   
450664  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...        /m/02w7zz8   
450665  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...        /m/02w7zz8   
450666  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...        /m/02w7zz8   
450667  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...        /m/02w7zz8   
450668  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...        /m/02w7zz8   

        ...               CharacterName    ActorDOB ActorGender ActorHeight  \
0       ...                    Akooshay  1958-08-26           F       1.620   
1       ...  Lieutenant Melanie Ballard  1974-08-15           F       1.780   
2       ...         Desolation Williams  1969-06-15           M       1.727   
3       ...          Sgt Jericho Butler  1967-09-12           M       1.750   
4       ...             Bashira Kincaid  1977-09-25           F       1.650   
...     ...                         ...         ...         ...         ...   
450664  ...                         NaN         NaN         NaN         NaN   


In [24]:
merged_data.columns

Index(['WikipediaMovieID', 'FreebaseMovieID_x', 'MovieName', 'ReleaseDate_x',
       'BoxOfficeRevenue', 'Runtime', 'Languages', 'Countries', 'Genres',
       'FreebaseMovieID_y', 'ReleaseDate_y', 'CharacterName', 'ActorDOB',
       'ActorGender', 'ActorHeight', 'ActorEthnicity', 'ActorName',
       'ActorAgeAtRelease', 'FreebaseCharacterActorMapID',
       'FreebaseCharacterID', 'FreebaseActorID'],
      dtype='object')

In [29]:
merged_data = merged_data.drop(columns=[col for col in merged_data.columns if col.endswith('_y')])
merged_data.columns = [col.replace('_x', '') for col in merged_data.columns]


In [30]:
merged_data

WikipediaMovieID FreebaseMovieID       MovieName ReleaseDate  \
0                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
1                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
2                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
3                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
4                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
...                  ...             ...             ...         ...   
450664          12476867      /m/02w7zz8         Spliced        2002   
450665          12476867      /m/02w7zz8         Spliced        2002   
450666          12476867      /m/02w7zz8         Spliced        2002   
450667          12476867      /m/02w7zz8         Spliced        2002   
450668          12476867      /m/02w7zz8         Spliced        2002   

        BoxOfficeRevenue  Runtime                           Languages  \
0             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
2             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
3             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
4             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
...                  ...      ...                                 ...   
450664               NaN     86.0  {"/m/02h40lc": "English Language"}   
450665               NaN     86.0  {"/m/02h40lc": "English Language"}   
450666               NaN     86.0  {"/m/02h40lc": "English Language"}   
450667               NaN     86.0  {"/m/02h40lc": "English Language"}   
450668               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                        Countries  \
0       {"/m/09c7w0": "United States of America"}   
1       {"/m/09c7w0": "United States of America"}   
2       {"/m/09c7w0": "United States of America"}   
3       {"/m/09c7w0": "United States of America"}   
4       {"/m/09c7w0": "United States of America"}   
...                                           ...   
450664                    {"/m/0d060g": "Canada"}   
450665                    {"/m/0d060g": "Canada"}   
450666                    {"/m/0d060g": "Canada"}   
450667                    {"/m/0d060g": "Canada"}   
450668                    {"/m/0d060g": "Canada"}   

                                                   Genres  \
0       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
1       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
2       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
3       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
4       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
...                                                   ...   
450664  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450665  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450666  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450667  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450668  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                         NaN         NaN         NaN          NaN   
450665                         NaN  1982-01-28         NaN          NaN   
450666                         NaN         NaN           F          NaN   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN

In [83]:
threshold = int(0.80 * merged_data.shape[1])

merged_data = merged_data.dropna(thresh=threshold)

In [84]:
merged_data

WikipediaMovieID FreebaseMovieID  \
0                 975900       /m/03vyhn   
1                 975900       /m/03vyhn   
2                 975900       /m/03vyhn   
3                 975900       /m/03vyhn   
4                 975900       /m/03vyhn   
...                  ...             ...   
450659            913762       /m/03pcrp   
450660            913762       /m/03pcrp   
450661          12476867      /m/02w7zz8   
450662          12476867      /m/02w7zz8   
450663          12476867      /m/02w7zz8   

                                                MovieName ReleaseDate  \
0                                          Ghosts of Mars  2001-08-24   
1                                          Ghosts of Mars  2001-08-24   
2                                          Ghosts of Mars  2001-08-24   
3                                          Ghosts of Mars  2001-08-24   
4                                          Ghosts of Mars  2001-08-24   
...                                                   ...         ...   
450659  The Super Dimension Fortress Macross II: Lover...  1992-05-21   
450660  The Super Dimension Fortress Macross II: Lover...  1992-05-21   
450661                                            Spliced        2002   
450662                                            Spliced        2002   
450663                                            Spliced        2002   

        BoxOfficeRevenue  Runtime                           Languages  \
0             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
2             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
3             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
4             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
...                  ...      ...                                 ...   
450659               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
450660               NaN    150.0   {"/m/03_9r": "Japanese Language"}   
450661               NaN     86.0  {"/m/02h40lc": "English Language"}   
450662               NaN     86.0  {"/m/02h40lc": "English Language"}   
450663               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                        Countries  \
0       {"/m/09c7w0": "United States of America"}   
1       {"/m/09c7w0": "United States of America"}   
2       {"/m/09c7w0": "United States of America"}   
3       {"/m/09c7w0": "United States of America"}   
4       {"/m/09c7w0": "United States of America"}   
...                                           ...   
450659                      {"/m/03_3d": "Japan"}   
450660                      {"/m/03_3d": "Japan"}   
450661                    {"/m/0d060g": "Canada"}   
450662                    {"/m/0d060g": "Canada"}   
450663                    {"/m/0d060g": "Canada"}   

                                                   Genres    year  \
0       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   
1       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   
2       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   
3       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   
4       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  2001.0   
...                                                   ...     ...   
450659  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  1992.0   
450660  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  1992.0   
450661  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  2002.0   
450662  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  2002.0   
450663  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  2002.0   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams

We lost more than 100k rows in 75% threshold and more than 200k in 80% threshold 

## Oscar dataset

In [97]:
OscarWithRatings=pd.read_csv('data/oscar-movies-ratings/versions/2/oscars_df.csv')
OscarWithRatings

Unnamed: 0                        Film Oscar Year  \
0             0                       Wings    1927/28   
1             1                  7th Heaven    1927/28   
2             2                  The Racket    1927/28   
3             3         The Broadway Melody    1928/29   
4             4                       Alibi    1928/29   
..          ...                         ...        ...   
566         566                        Mank       2020   
567         567                      Minari       2020   
568         568       Promising Young Woman       2020   
569         569              Sound of Metal       2020   
570         570  The Trial of the Chicago 7       2020   

                               Film Studio/Producer(s)    Award  \
0                                 Famous Players-Lasky   Winner   
1                                                  Fox  Nominee   
2                                    The Caddo Company  Nominee   
3                                  Metro-Goldwyn-Mayer   Winner   
4                                  Feature Productions  Nominee   
..                                                 ...      ...   
566      Ceán Chaffin, Eric Roth, and Douglas Urbanski  Nominee   
567                                       Christina Oh  Nominee   
568  Ben Browning, Ashley Fox, Emerald Fennell, and...  Nominee   
569              Bert Hamelinck and Sacha Ben Harroche  Nominee   
570                    Marc Platt and Stuart M. Besser  Nominee   

     Year of Release  Movie Time             Movie Genre  IMDB Rating  \
0               1927         144       Drama,Romance,War          7.5   
1               1927         110           Drama,Romance          7.7   
2               1928          84   Crime,Drama,Film-Noir          6.7   
3               1929         100   Drama,Musical,Romance          5.7   
4               1929          91    Action,Crime,Romance          5.8   
..               ...         ...                     ...          ...   
566             2020         131  Biography,Comedy,Drama          6.9   
567             2020         115                   Drama          7.5   
568             2020         113    Crime,Drama,Thriller          7.5   
569             2019         120             Drama,Music          7.8   
570             2020         129  Drama,History,Thriller          7.8   

    IMDB Votes  ... Tomatometer Status Tomatometer Rating Tomatometer Count  \
0       12,221  ...    Certified-Fresh               93.0              46.0   
1        3,439  ...                NaN                NaN               NaN   
2        1,257  ...                NaN                NaN               NaN   
3        6,890  ...             Rotten               33.0              24.0   
4          765  ...                NaN                NaN               NaN   
..         ...  ...                ...                ...               ...   
566     65,380  ...                NaN                NaN               NaN   
567     57,976  ...                NaN                NaN               NaN   
568    122,269  ...                NaN                NaN               NaN   
569    102,807  ...                NaN                NaN               NaN   
570    156,283  ...                NaN                NaN               NaN   

    Audience Status Audience Rating Audience Count  \
0           Upright            78.0         3530.0   
1               NaN             NaN            NaN   
2               NaN             NaN            NaN   
3           Spilled            21.0         1813.0   
4               NaN             NaN            NaN   
..              ...             ...            ...   
566             NaN             NaN            NaN   
567             NaN             NaN            NaN   
568             NaN             NaN            NaN   
569             NaN             NaN            NaN   
570             NaN             NaN            NaN   

    Tomatometer Top Critics 

In [98]:
OscarWithRatings.columns

Index(['Unnamed: 0', 'Film', 'Oscar Year', 'Film Studio/Producer(s)', 'Award',
       'Year of Release', 'Movie Time', 'Movie Genre', 'IMDB Rating',
       'IMDB Votes', 'Movie Info', 'Genres', 'Critic Consensus',
       'Content Rating', 'Directors', 'Authors', 'Actors',
       'Original Release Date', 'Streaming Release Date', 'Production Company',
       'Tomatometer Status', 'Tomatometer Rating', 'Tomatometer Count',
       'Audience Status', 'Audience Rating', 'Audience Count',
       'Tomatometer Top Critics Count', 'Tomatometer Fresh Critics Count',
       'Tomatometer Rotten Critics Count', 'Film ID'],
      dtype='object')

In [99]:
oscar_awards=pd.read_csv('data/the-oscar-award/versions/11/the_oscar_award.csv')

oscar_awards

year_film  year_ceremony  ceremony                          category  \
0           1927           1928         1                             ACTOR   
1           1927           1928         1                             ACTOR   
2           1927           1928         1                           ACTRESS   
3           1927           1928         1                           ACTRESS   
4           1927           1928         1                           ACTRESS   
...          ...            ...       ...                               ...   
10884       2023           2024        96     WRITING (Original Screenplay)   
10885       2023           2024        96  JEAN HERSHOLT HUMANITARIAN AWARD   
10886       2023           2024        96                    HONORARY AWARD   
10887       2023           2024        96                    HONORARY AWARD   
10888       2023           2024        96                    HONORARY AWARD   

                                                    name              film  \
0                                    Richard Barthelmess         The Noose   
1                                          Emil Jannings  The Last Command   
2                                         Louise Dresser   A Ship Comes In   
3                                           Janet Gaynor        7th Heaven   
4                                         Gloria Swanson    Sadie Thompson   
...                                                  ...               ...   
10884                             Written by Celine Song        Past Lives   
10885                                                NaN               NaN   
10886  To Angela Bassett, who has inspired audiences ...               NaN   
10887  To Mel Brooks, for his comedic brilliance, pro...               NaN   
10888  To Carol Littleton, whose commitment to her cr...               NaN   

       winner  
0       False  
1        True  
2       False  
3        True  
4       False  
...       ...  
10884   False  
10885    True  
10886    True  
10887    True  
10888    True  

[10889 rows x 7 columns]

In [100]:
oscar_awards['actor']=oscar_awards['category'].str.contains(r'actor|actress', case=False)

df_actors=oscar_awards[oscar_awards['actor']==True]
df_actors

year_film  year_ceremony  ceremony                      category  \
0           1927           1928         1                         ACTOR   
1           1927           1928         1                         ACTOR   
2           1927           1928         1                       ACTRESS   
3           1927           1928         1                       ACTRESS   
4           1927           1928         1                       ACTRESS   
...          ...            ...       ...                           ...   
10780       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   
10781       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   
10782       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   
10783       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   
10784       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   

                       name              film  winner  actor  
0       Richard Barthelmess         The Noose   False   True  
1             Emil Jannings  The Last Command    True   True  
2            Louise Dresser   A Ship Comes In   False   True  
3              Janet Gaynor        7th Heaven    True   True  
4            Gloria Swanson    Sadie Thompson   False   True  
...                     ...               ...     ...    ...  
10780           Emily Blunt       Oppenheimer   False   True  
10781       Danielle Brooks  The Color Purple   False   True  
10782       America Ferrera            Barbie   False   True  
10783          Jodie Foster              Nyad   False   True  
10784  Da'Vine Joy Randolph     The Holdovers    True   True  

[1828 rows x 8 columns]

In [101]:
df_actors['category'].unique()

array(['ACTOR', 'ACTRESS', 'ACTOR IN A SUPPORTING ROLE',
       'ACTRESS IN A SUPPORTING ROLE', 'ACTOR IN A LEADING ROLE',
       'ACTRESS IN A LEADING ROLE'], dtype=object)

In [102]:
df_actors['winner'].sum()/len(df_actors)

0.2024070021881838

In [103]:
df_oscar_winners=df_actors[df_actors['winner']==True].drop(columns='actor')
df_oscar_winners

year_film  year_ceremony  ceremony                      category  \
1           1927           1928         1                         ACTOR   
3           1927           1928         1                       ACTRESS   
36          1928           1929         2                         ACTOR   
45          1928           1929         2                       ACTRESS   
73          1929           1930         3                         ACTOR   
...          ...            ...       ...                           ...   
10657       2022           2023        95  ACTRESS IN A SUPPORTING ROLE   
10768       2023           2024        96       ACTOR IN A LEADING ROLE   
10772       2023           2024        96    ACTOR IN A SUPPORTING ROLE   
10779       2023           2024        96     ACTRESS IN A LEADING ROLE   
10784       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   

                       name                               film  winner  
1             Emil Jannings                   The Last Command    True  
3              Janet Gaynor                         7th Heaven    True  
36            Warner Baxter                     In Old Arizona    True  
45            Mary Pickford                           Coquette    True  
73            George Arliss                           Disraeli    True  
...                     ...                                ...     ...  
10657      Jamie Lee Curtis  Everything Everywhere All at Once    True  
10768        Cillian Murphy                        Oppenheimer    True  
10772     Robert Downey Jr.                        Oppenheimer    True  
10779            Emma Stone                        Poor Things    True  
10784  Da'Vine Joy Randolph                      The Holdovers    True  

[370 rows x 7 columns]

In [104]:
len(df_oscar_winners['name'].unique())

318

In [105]:
df_oscar_winners['actor_clean_name'] = df_oscar_winners['name'].str.lower().str.strip().str.replace(r'[^\w\s]', '', regex=True)
df_oscar_winners['movie_clean_name']=df_oscar_winners['film'].str.lower().str.strip().str.replace(r'[^\w\s]', '', regex=True)
df_oscar_winners

year_film  year_ceremony  ceremony                      category  \
1           1927           1928         1                         ACTOR   
3           1927           1928         1                       ACTRESS   
36          1928           1929         2                         ACTOR   
45          1928           1929         2                       ACTRESS   
73          1929           1930         3                         ACTOR   
...          ...            ...       ...                           ...   
10657       2022           2023        95  ACTRESS IN A SUPPORTING ROLE   
10768       2023           2024        96       ACTOR IN A LEADING ROLE   
10772       2023           2024        96    ACTOR IN A SUPPORTING ROLE   
10779       2023           2024        96     ACTRESS IN A LEADING ROLE   
10784       2023           2024        96  ACTRESS IN A SUPPORTING ROLE   

                       name                               film  winner  \
1             Emil Jannings                   The Last Command    True   
3              Janet Gaynor                         7th Heaven    True   
36            Warner Baxter                     In Old Arizona    True   
45            Mary Pickford                           Coquette    True   
73            George Arliss                           Disraeli    True   
...                     ...                                ...     ...   
10657      Jamie Lee Curtis  Everything Everywhere All at Once    True   
10768        Cillian Murphy                        Oppenheimer    True   
10772     Robert Downey Jr.                        Oppenheimer    True   
10779            Emma Stone                        Poor Things    True   
10784  Da'Vine Joy Randolph                      The Holdovers    True   

          actor_clean_name                   movie_clean_name  
1            emil jannings                   the last command  
3             janet gaynor                         7th heaven  
36           warner baxter                     in old arizona  
45           mary pickford                           coquette  
73           george arliss                           disraeli  
...                    ...                                ...  
10657     jamie lee curtis  everything everywhere all at once  
10768       cillian murphy                        oppenheimer  
10772     robert downey jr                        oppenheimer  
10779           emma stone                        poor things  
10784  davine joy randolph                      the holdovers  

[370 rows x 9 columns]

In [106]:
merged_data=pd.merge(movie_metadata,character_metadata,how='inner',on='WikipediaMovieID')
merged_data = merged_data.drop(columns=[col for col in merged_data.columns if col.endswith('_y')])
merged_data.columns = [col.replace('_x', '') for col in merged_data.columns]
merged_data

WikipediaMovieID FreebaseMovieID       MovieName ReleaseDate  \
0                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
1                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
2                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
3                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
4                 975900       /m/03vyhn  Ghosts of Mars  2001-08-24   
...                  ...             ...             ...         ...   
450664          12476867      /m/02w7zz8         Spliced        2002   
450665          12476867      /m/02w7zz8         Spliced        2002   
450666          12476867      /m/02w7zz8         Spliced        2002   
450667          12476867      /m/02w7zz8         Spliced        2002   
450668          12476867      /m/02w7zz8         Spliced        2002   

        BoxOfficeRevenue  Runtime                           Languages  \
0             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
2             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
3             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
4             14010832.0     98.0  {"/m/02h40lc": "English Language"}   
...                  ...      ...                                 ...   
450664               NaN     86.0  {"/m/02h40lc": "English Language"}   
450665               NaN     86.0  {"/m/02h40lc": "English Language"}   
450666               NaN     86.0  {"/m/02h40lc": "English Language"}   
450667               NaN     86.0  {"/m/02h40lc": "English Language"}   
450668               NaN     86.0  {"/m/02h40lc": "English Language"}   

                                        Countries  \
0       {"/m/09c7w0": "United States of America"}   
1       {"/m/09c7w0": "United States of America"}   
2       {"/m/09c7w0": "United States of America"}   
3       {"/m/09c7w0": "United States of America"}   
4       {"/m/09c7w0": "United States of America"}   
...                                           ...   
450664                    {"/m/0d060g": "Canada"}   
450665                    {"/m/0d060g": "Canada"}   
450666                    {"/m/0d060g": "Canada"}   
450667                    {"/m/0d060g": "Canada"}   
450668                    {"/m/0d060g": "Canada"}   

                                                   Genres  \
0       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
1       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
2       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
3       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
4       {"/m/01jfsb": "Thriller", "/m/06n90": "Science...   
...                                                   ...   
450664  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450665  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450666  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450667  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   
450668  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...   

                     CharacterName    ActorDOB ActorGender  ActorHeight  \
0                         Akooshay  1958-08-26           F        1.620   
1       Lieutenant Melanie Ballard  1974-08-15           F        1.780   
2              Desolation Williams  1969-06-15           M        1.727   
3               Sgt Jericho Butler  1967-09-12           M        1.750   
4                  Bashira Kincaid  1977-09-25           F        1.650   
...                            ...         ...         ...          ...   
450664                         NaN         NaN         NaN          NaN   
450665                         NaN  1982-01-28         NaN          NaN   
450666                         NaN         NaN           F          NaN   
450667                         NaN         NaN         NaN          NaN   
450668                         NaN         NaN         NaN          NaN

In [107]:
merged_data['actor_clean_name']=merged_data['ActorName'].str.lower().str.strip().str.replace(r'[^\w\s]', '', regex=True)
merged_data['movie_clean_name']=merged_data['MovieName'].str.lower().str.strip().str.replace(r'[^\w\s]', '', regex=True)


In [108]:
merged_data=pd.merge(merged_data,df_oscar_winners,how='outer',left_on=['actor_clean_name', 'movie_clean_name'], right_on=['actor_clean_name', 'movie_clean_name'])

In [109]:
merged_data

WikipediaMovieID FreebaseMovieID               MovieName ReleaseDate  \
0             11901968.0      /m/02rxcn3       Big Money Rustlas        2010   
1             21029252.0      /m/05b3f51              Miss March  2009-03-13   
2             21798180.0      /m/05mz_dh                      13  2010-03-13   
3              6501095.0       /m/0g7sfc  50 Cent: The New Breed  2003-04-15   
4             33638321.0      /m/0h28lbt   All Things Fall Apart        2011   
...                  ...             ...                     ...         ...   
450716         6415208.0       /m/0g4fl5               Zachariah  1971-01-24   
450717        34953010.0      /m/0j43swk        Zero Dark Thirty  2012-12-19   
450718        34953010.0      /m/0j43swk        Zero Dark Thirty  2012-12-19   
450719         4463641.0       /m/0c3vzx         Zombie Genocide        1993   
450720         8720274.0      /m/027g6y5                     Zoo  2007-04-25   

        BoxOfficeRevenue  Runtime  \
0                    NaN     95.0   
1              4591629.0     90.0   
2                    NaN     97.0   
3                    NaN      NaN   
4                    NaN      NaN   
...                  ...      ...   
450716               NaN     92.0   
450717               NaN      NaN   
450718               NaN      NaN   
450719               NaN     64.0   
450720               NaN     76.0   

                                                Languages  \
0                      {"/m/02h40lc": "English Language"}   
1       {"/m/05zjd": "Portuguese Language", "/m/02h40l...   
2                      {"/m/02h40lc": "English Language"}   
3                      {"/m/02h40lc": "English Language"}   
4                      {"/m/02h40lc": "English Language"}   
...                                                   ...   
450716                 {"/m/02h40lc": "English Language"}   
450717                 {"/m/02h40lc": "English Language"}   
450718                 {"/m/02h40lc": "English Language"}   
450719                 {"/m/02h40lc": "English Language"}   
450720                 {"/m/02h40lc": "English Language"}   

                                        Countries  \
0       {"/m/09c7w0": "United States of America"}   
1       {"/m/09c7w0": "United States of America"}   
2       {"/m/09c7w0": "United States of America"}   
3       {"/m/09c7w0": "United States of America"}   
4       {"/m/09c7w0": "United States of America"}   
...                                           ...   
450716  {"/m/09c7w0": "United States of America"}   
450717  {"/m/09c7w0": "United States of America"}   
450718  {"/m/09c7w0": "United States of America"}   
450719                    {"/m/03rt9": "Ireland"}   
450720  {"/m/09c7w0": "United States of America"}   

                                                   Genres       CharacterName  \
0       {"/m/0gf28": "Parody", "/m/0lsxr": "Crime Fict...                 NaN   
1       {"/m/04228s": "Road movie", "/m/0gsy3b": "Sex ...                 NaN   
2        {"/m/01jfsb": "Thriller", "/m/07s9rl0": "Drama"}               Jimmy   
3       {"/m/04rlf": "Music", "/m/0d2rhq": "Concert fi...                 NaN   
4         {"/m/01z02hx": "Sports", "/m/07s9rl0": "Drama"}                Deon   
...                                                   ...                 ...   
450716  {"/m/0219x_": "Indie", "/m/04t36": "Musical", ...  Belle Starr's Band   
450717  {"/m/01jfsb": "Thriller", "/m/02kdv5l": "Actio...                 NaN   
450718  {"/m/01jfsb": "Thriller", "/m/02kdv5l": "Actio...                 NaN   
450719                      {"/m/0jb4p32": "Zombie Film"}                 NaN   
450720  {"/m/0219x_": "Indie", "/m/0jtdp": "Documentary"}                 NaN   

        ... FreebaseActorID actor_clean_name       movie_clean_name year_film  \
0       ...      /m/0421n4b      2 tuff tony      big money rustlas       NaN   
1       ...       /m/09rsqv         40 glocc             miss march       NaN   
2       ...     

In [110]:
merged_data.columns

Index(['WikipediaMovieID', 'FreebaseMovieID', 'MovieName', 'ReleaseDate',
       'BoxOfficeRevenue', 'Runtime', 'Languages', 'Countries', 'Genres',
       'CharacterName', 'ActorDOB', 'ActorGender', 'ActorHeight',
       'ActorEthnicity', 'ActorName', 'ActorAgeAtRelease',
       'FreebaseCharacterActorMapID', 'FreebaseCharacterID', 'FreebaseActorID',
       'actor_clean_name', 'movie_clean_name', 'year_film', 'year_ceremony',
       'ceremony', 'category', 'name', 'film', 'winner'],
      dtype='object')

In [111]:
merged_data[merged_data['winner']==True]

WikipediaMovieID FreebaseMovieID                     MovieName  \
3382           1897341.0       /m/064lsn                   The Pianist   
6158            133648.0        /m/0_9wr              Scent of a Woman   
6565           7047921.0       /m/0h1x5f          Little Miss Sunshine   
8728             42856.0        /m/0bs4r  The Bridge on the River Kwai   
11909            61527.0        /m/0gns4                In Old Chicago   
...                  ...             ...                           ...   
438632               NaN             NaN                           NaN   
440308          156641.0       /m/014knw                     Stalag 17   
440432          113452.0        /m/0sxkh      Kiss of the Spider Woman   
445273               NaN             NaN                           NaN   
445581         4015886.0       /m/0bcndz                The King and I   

       ReleaseDate  BoxOfficeRevenue  Runtime  \
3382    2002-05-24       120072577.0    142.0   
6158    1992-12-23               NaN    137.0   
6565    2006-01-20               NaN    102.0   
8728    1957-10-02        33300000.0    155.0   
11909         1938               NaN     94.0   
...            ...               ...      ...   
438632         NaN               NaN      NaN   
440308        1953         3300000.0    120.0   
440432  1985-05-13        17005229.0    120.0   
445273         NaN               NaN      NaN   
445581  1956-06-28         8500000.0    133.0   

                                                Languages  \
3382    {"/m/06b_j": "Russian Language", "/m/04306rv":...   
6158                   {"/m/02h40lc": "English Language"}   
6565                   {"/m/02h40lc": "English Language"}   
8728    {"/m/03_9r": "Japanese Language", "/m/02h40lc"...   
11909                  {"/m/02h40lc": "English Language"}   
...                                                   ...   
438632                                                NaN   
440308                 {"/m/02h40lc": "English Language"}   
440432  {"/m/064_8sq": "French Language", "/m/02h40lc"...   
445273                                                NaN   
445581  {"/m/02h40lc": "English Language", "/m/0c_v2":...   

                                                Countries  \
3382    {"/m/0f8l9c": "France", "/m/05qhw": "Poland", ...   
6158            {"/m/09c7w0": "United States of America"}   
6565            {"/m/09c7w0": "United States of America"}   
8728    {"/m/09c7w0": "United States of America", "/m/...   
11909           {"/m/09c7w0": "United States of America"}   
...                                                   ...   
438632                                                NaN   
440308          {"/m/09c7w0": "United States of America"}   
440432  {"/m/09c7w0": "United States of America", "/m/...   
445273                                                NaN   
445581          {"/m/09c7w0": "United States of America"}   

                                                   Genres  \
3382    {"/m/03g3w": "History", "/m/03bxz7": "Biograph...   
6158    {"/m/07s9rl0": "Drama", "/m/01j1n2": "Coming o...   
6565    {"/m/0hn10": "LGBT", "/m/0hj3n0w": "Ensemble F...   
8728    {"/m/0hj3n0w": "Ensemble Film", "/m/07s9rl0": ...   
11909   {"/m/01drsx": "Disaster", "/m/03btsm8": "Actio...   
...                                                   ...   
438632                                                NaN   
440308  {"/m/0hj3n0w": "Ensemble Film", "/m/01t_vv": "...   
440432  {"/m/0hn10": "LGBT", "/m/0lsxr": "Crime Fictio...   
445273                                                NaN   
445581  {"/m/04xvh5": "Costume drama", "/m/04xvlr": "P...   

               CharacterName  ... FreebaseActorID actor_clean_name  \
3382      Wladyslaw Szpilman  ...       /m/01cj6y     adrien brody   
6158    Lt. Col. Frank Slade  ...        /m/0bj9k        al pacino   
6565    Grandpa Edwin Hoover  ...       /m/015grj       alan arkin   
8728       Colonel Nicholson  ...        /m/0gr36  

In [112]:
duplicate_movies =merged_data[merged_data.duplicated(subset=['movie_clean_name', 'actor_clean_name', 'winner'], keep=False)]
print(duplicate_movies)


        WikipediaMovieID FreebaseMovieID                         MovieName  \
290           18999254.0      /m/02z6zks              Onks' Viljoo näkyny?   
291           18999254.0      /m/02z6zks              Onks' Viljoo näkyny?   
292           18999254.0      /m/02z6zks              Onks' Viljoo näkyny?   
293           18999254.0      /m/02z6zks              Onks' Viljoo näkyny?   
294           18999254.0      /m/02z6zks              Onks' Viljoo näkyny?   
...                  ...             ...                               ...   
450702         8326198.0      /m/026_hnv  Who the Fuck Is Jackson Pollock?   
450704        27947216.0      /m/0ch185_                       Witchhammer   
450705        27947216.0      /m/0ch185_                       Witchhammer   
450717        34953010.0      /m/0j43swk                  Zero Dark Thirty   
450718        34953010.0      /m/0j43swk                  Zero Dark Thirty   

       ReleaseDate  BoxOfficeRevenue  Runtime  \
290     1988-1